src/client.rs

```diff
@@ -0,0 +1,850 @@
+use jsonrpc::error::Error;
+
+use super::primitives;
+
+pub struct Client {
+	agent: jsonrpc::client::Client
```

Comment:
In my opinion, it looks a bit cleaner to import this under a different name, i.e.,
`use jsonrpc::client::Client as RpcClient;`
That makes the code a bit more readable.